server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -1996,6 +1996,23 @@ private RemoteSegmentStoreDirectory getRemoteDirectory() {
         return ((RemoteSegmentStoreDirectory) remoteDirectory);
     }
 
+    /*
+    Returns true iff it is able to verify that there is at least one
+    remote segment metadata uploaded
+     */
+    boolean atLeastOneRemoteSync() {
```

Comment:
Can we also check if the segment files are same? `RemoteSegmentStoreDirectory.getSegmentsUploadedToRemoteStore` will provide details of segments uploaded to remote store and `store.directory().listAll()` will give local segments info.

It would be a better check as long as it does not create false positives

Reply:
Makes sense . Added it .